common/hyperscaler/rules/rule.go

```diff
@@ -0,0 +1,224 @@
+package rules
+
+import (
+	"fmt"
+	"sort"
+)
+
+type Labels struct {
+	Labels []string
+}
+
+type Rule struct {
+	Plan                           string
+	PlatformRegion                 string
+	HyperscalerRegion              string
+	EuAccess                       bool
+	Shared                         bool
+	Labels                         map[string]string
+	ContainsInputAttributes        bool
+	ContainsOutputAttributes       bool
+	hyperscalerNameMappingFunction func(string) string
+}
+
+func NewRule() *Rule {
+	return &Rule{
+		hyperscalerNameMappingFunction: getHyperscalerName,
+		Labels:                         make(map[string]string),
+	}
+}
+
+type MatchableAttributes struct {
+	Plan              string `json:"plan"`
+	PlatformRegion    string `json:"platformRegion"`
+	HyperscalerRegion string `json:"hyperscalerRegion"`
+}
+
+func (r *Rule) CalculateLabels() map[string]string {
+	return r.CalculateLabelsWith(getHyperscalerName(r.Plan))
+}
+
+func (r *Rule) CalculateLabelsWith(hyperscalerName string) map[string]string {
+	for _, attr := range AllAttributes {
+		if attr.Getter(r) != "" {
```

Comment:
Could we use some attribute o `attr` with boolean value or even create helper function returning all non-empty attributes?

Reply:
It should be clear and simple for everyone trying to read it/modify it.

Reply:
Feel free to correct it as you wish.